test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts

```diff
@@ -157,6 +173,13 @@ contract('DCAStrategiesManagementHandler', () => {
         );
       });
     });
+    when('token shares exceed max amount', () => {
```

Comment:
Maybe something like this can help to avoid repeating the same tests:
https://github.com/Mean-Finance/dca-v2-periphery/blob/main/test/unit/DCAHubCompanion/dca-hub-companion-hub-proxy-handler.spec.ts#L224

The idea is basically to have a function like:
```
function invalidSharesTest({ method, params }: { method: string, params: (shares) => any[] }) {
    when('token shares exceed max amount', () => {
        then('tx reverted with message', async () => {
            const invalidParams = params(SHARES_EXCEED_AMOUNT)
            await expect(DCAStrategiesManagementHandlerMock.connect(user)[method](invalidParams)).to.be.revertedWith(
          'TokenSharesExceedAmount()'
            );
         });
    });
    }
}
```

And then you can call:
```
invalidSharesTest({ method: 'createStrategy', params: (shares) => [NAME, shares, user.address]})
invalidSharesTest({ method: 'updateStrategyTokens', params: (shares) => [1, shares]})

Reply:
Works perfectly!